src/pages/workspace/WorkspaceMoreFeaturesPage.tsx

```diff
@@ -88,6 +122,24 @@ function WorkspaceMoreFeaturesPage({policy, route}: WorkspaceMoreFeaturesPagePro
         },
     ];
 
+    // TODO remove this when feature will be fully done, and move spend item inside spendItems array
+    if (canUseWorkspaceFeeds) {
+        spendItems.splice(1, 0, {
+            icon: Illustrations.HandCard,
+            titleTranslationKey: 'workspace.moreFeatures.expensifyCard.title',
+            subtitleTranslationKey: 'workspace.moreFeatures.expensifyCard.subtitle',
+            isActive: policy?.areExpensifyCardsEnabled ?? false,
+            pendingAction: policy?.pendingFields?.areExpensifyCardsEnabled,
+            disabled: !isEmptyObject(cardsList),
+            action: (isEnabled: boolean) => {
+                Policy.enableExpensifyCard(policy?.id ?? '-1', isEnabled);
```

Comment:
What does is expected to happen if we call `Policy.enableExpensifyCard` with `policyID = -1`? I guess the server will throw an error... shouldn't we instead be writing the code like:

```
action: (isEnabled: boolean) => {
  if (!policy?.id) {
    return;
  }
  Policy.enableExpensifyCard(policy.id, isEnabled);
}
```
and disable this button if we don't have a `policy.id` with `disabled: !isEmptyObject(cardsList) || policy?.id,`

Reply:
Good point, @aldo-expensify! 

@narefyev91 Can we update to not call Policy.enableExpensifyCard with policyID `-1`?

Reply:
sure

Reply:
BTW @MariaHCD @aldo-expensify we have the same logic for all menu items
<img width="749" alt="Screenshot 2024-08-05 at 13 49 49" src="https://github.com/user-attachments/assets/6bae17ec-f5de-42b2-950d-20d54af32b46">
And probably all those API calls will failed?

Reply:
A little bit confusing if we just re-write only one menu item - and leave everything else as it is. And why such behaviour was not implemented? Probably we do not block action even if policy id is not exists?

Reply:
> And probably all those API calls will failed?

Oh, I didn't realize it was a pattern that was used in the other API calls. But yes, those API calls will all fail because the policyID `-1` is invalid.

Perhaps let's update all of those to not make the API calls if the policyID is not present?

Reply:
Probably good ask - but we do not have such logic anywhere in the app - that if policy id is not exists - we block user interaction. Probably we need some approvals from @mountiny or some one from top architectural stand point

Reply:
> And why such behaviour was not implemented? Probably we do not block action even if policy id is not exists?

My guess is that the `policy.id` is really never missing at this point so maybe no one has seen this causing an error. My suggestion is more on the lines of writing code that makes more sense when someone reads it, and also in case someone manages to click these buttons when the policy has not loaded (if if if that can really happen)

This looks like a bad use of coalesce `?? -1` just to write less code.

> but we do not have such logic anywhere in the app - that if policy id is not exists - we block user interaction

Are you suggesting that we may want to allow the interaction in this case? if you allow the interaction while the policy is missing, it is for sure just going to cause an error that is probably not handled gracefully, why would we want that?

Reply:
@aldo-expensify i fully agree with you - that if we do not have policy id - we should just not allow interaction
Yeah i think we allowing interaction in 99% cases - no matter if it will be error or success - that's probably why we have in onyx both optimistic, success and error cases pre-defined.
Long story short - i will glad to add those checks for not allowing interaction with toggles - if policy id does not exists.
@trjExpensify @mountiny @MariaHCD @aldo-expensify can you please give a green light here?

Reply:
🟢 from me

From a backend perspective, I think it makes sense not to call the API unnecessarily when we do not have a policyID - those API calls would just fail - and so would be redundant.

Happy to hear more thoughts on the UX, tho. If the policyID is not present and the call fails, the toggle would just flip back to it's previous state which might be more confusing UX than if the toggle was not interactable at all.

Reply:
@shawnborton as the UX expert - your thoughts?

Reply:
Can you summarize the situation in which you are asking for my thoughts? I'm not quite following what we're talking about here.

Reply:
@shawnborton Problem - we allow to click on any toggle in "More sections" if policy is not exists (for some reason)
<img width="490" alt="Screenshot 2024-08-07 at 14 34 00" src="https://github.com/user-attachments/assets/88462cbb-0da1-4c81-9b51-e086f85c7406">
We have 3 options here:
1) disable toggle - not give user ability to off/on
2) give user ability to click on toggle (off/on) - but if API failed move toggle state back (on/off) - which will be confusing UX
3) don't do anything here

We currently discussing to fully block user interaction if policy is not exists.

Reply:
How can you access this screen if the policy doesn't exist though?

Reply:
Good question - probably not possible based on [src/pages/workspace/withPolicyAndFullscreenLoading.tsx](https://github.com/Expensify/App/blob/main/src/pages/workspace/withPolicyAndFullscreenLoading.tsx)

Reply:
Got it. I'm not quite following what the situation is exactly that you need advice on, but if I can be helpful, please let me know and we can keep discussing.

Reply:
I think this matters more from a code perspective rather than UI, since it may be impossible for this to happen in a normal user case? 🤷

Reply:
Ok I will make that changes and push a PR to clean up code

Reply:
Wanted to say that this screen has the policyID in the URL so probably not possible to access.

I do like the clarity though from the code perspective so feel free to clean this up but noting this is lower priority task so try to prioritize other release related work and. Focus on this if you do not have other tasks to handle @narefyev91 thanks!

Thanks @aldo-expensify for raising

Reply:
@mountiny yeah - i'm happy to help with anything for upcoming releases - but probably for now have some time to pick this one (until something else will come up). Probably waiting when Third-party workspace card feed doc will reviewed and start on that project